docs/design/backupstoragelocationserver-design.md

```diff
@@ -0,0 +1,214 @@
+
+The repository itself is defined in a separate `BackupStorageLocationRepository` (BSLR) custom resource and updated by the OADP BSLR controller. This decouples backend storage management from user access, allowing the BSLS to serve users securely without exposing storage credentials or infrastructure details.
+
+## Detailed Design
+
+### BSLS Overview
+
+* **Pointer to Repository**:  
+  Each `BackupStorageLocationServer` (BSLS) instance points to a single repository specified by a `BackupStorageLocationRepository` (BSLR) resource, which must be in a ready state for the BSLS to operate.
+
+* **Credential Management**:
+  * **User Access**:  
+    - User credentials (username/password) for accessing the BSLS are stored as OpenShift `Secrets`. These are managed by the BSLS controller and are used by Kopia clients running inside VMs.
+    - One BSLS can be shared between multiple users. These users cannot see each other’s snapshots, policies, or data. Access is isolated per user through Kopia’s access control mechanisms.
+
+  * **Repository Access**:
+    Credentials required by the BSLS to open and manage the Kopia repository are also stored as OpenShift `Secrets` and referenced in the OADP `DataProtectionApplication` (DPA) or `BSLR` CRDs. The BSLS uses these to authenticate to the backend storage securely. 
+* **Configuration**: 
+  - BSLS does not contain Kopia specific configuration parameters such as encryption algorithms, compression settings, and other repository-specific options, those are managed by the BSLR or DPA.
+  
+
+### BSLS Controller Responsibilities
+
+The BSLS controller is responsible for managing the lifecycle and configuration of the `BackupStorageLocationServer` resource and ensuring secure, multi-user access to the underlying Kopia repository.
+
+Key responsibilities include:
+
+ * **Repository Connectivity**: Ensure the associated `BackupStorageLocationRepository` (BSLR) is available and ready before exposing the server to users.
+
+  * **TLS and Network Security**: Configure TLS certificates for encrypted client-server communication. Ensure secure exposure of the Kopia API within the cluster network.
+
+ * **User Provisioning**: Reconcile Secrets or ConfigMaps containing user credentials and sync them to the Kopia repository to create or update user accounts.
+ 
+
+### Reconciliation Flow
+
+When a BackupStorageLocationServer (BSLS) resource is created or modified, the controller takes the following steps:
+
+ 1. **Validation**
+   * Verify that the spec.LocationRepository field references a valid and Ready BackupStorageLocationRepository (BSLR) in the same namespace.
+   * If invalid, mark the BSLS as NotReady and Requeue.
+ 2. **TLS Setup**
+   * Generate new or use a TLS certificate(s) from mounted from the OpenShift Secret for the BSLS service.
```

Comment:
We will need to make sure that anything we do here is FIPS-compliant if we generate the certs. I don't see why that would be an immediate problem, but it's something to verify.